pkg/odo/cli/component/devfile.go

```diff
@@ -169,16 +168,7 @@ func (do *DeleteOptions) DevfileComponentDelete() error {
 		return err
 	}
 
-	spinner := log.Spinner(fmt.Sprintf("Deleting devfile component %s", componentName))
-	defer spinner.End(false)
-
-	err = devfileHandler.Delete(labels)
-	if err != nil {
-		return err
-	}
-	spinner.End(true)
-	log.Successf("Successfully deleted component")
-	return nil
```

Comment:
These actions seems to be repeated in the delete call for both kube and docker. Any reason for removing it from here?

Reply:
I think its a bad UX if we're not deleting anything(when there is no component on `odo delete`) but showing `Deleting devfile component testsp11`/`Successfully deleted component` spinners.

Thats why I had the component exist check in this file initially, rather than the individual adapters. But since we decided to keep the component check on delete in the individual adapters, I think we should only show spinners when we're actually deleting the component.

Reply:
check https://github.com/openshift/odo/pull/3405#issuecomment-651913789 for the `odo delete` output from this PR

Reply:
But the problem is now the exist call is not covered by the spiners. Thus on a slow connection/cluster, that call might look unresponsive. Too many spiners is also a bad design. I think we can consider the exist call as a part of the delete action as it was called due to the delete process.

Reply:
updated, added the spinners to include the component check in their individual adapters